File: 6 Kyu/6 kyu - Esolang Interpreters #1 - Introduction to Esolangs and My First Interpreter (MiniStringFuck).cpp
std::string my_first_interpreter(const std::string& code) {

    std::string os;
    int c = 0;

    for (auto &it : code)
        if (it == '+' || it == '.') {
            if (c == 256) c = 0;
            if (it == '.') os += c;
            else c++;
        }

    return os;
}

File: 6 Kyu/6 kyu - Esolang Interpreters #1 - Introduction to Esolangs and My First Interpreter (MiniStringFuck).rs
fn my_first_interpreter(code: &str) -> String {
    
    let mut os = String::new();
    let mut mem = 0u8;
    
    for c in code.chars() {  
        match c {            
            '+' => mem = mem.wrapping_add (1),        
            '.' => os.push (mem as char),
             _  => ()       
        }      
    }
    
    os
}
